src/e2e/java/teammates/e2e/pageobjects/InstructorCourseEditPage.java

```diff
@@ -299,6 +299,7 @@ private void clickAddNewInstructorButton() {
 
     private void clickEditInstructorButton(int instrNum) {
         click(getEditInstructorButton(instrNum));
+        waitForPageToLoad(true);
```

Comment:
What event is being waited for here?

Reply:
The custom privilege settings panel is removed from the DOM when not in edit mode. The wait is for the custom privilege settings panel to be loaded as the test is looking for the custom privilege settings panel immediately after clicking the edit button.

Reply:
Then it's not exactly accurate to use `waitForPageToLoad`? It seems that the element to be waited here is very specific.

Reply:
That is very true. 
- Changed to `waitForPrivilegePanel` and added it at the specific locations where privilege panel is expected since it only exist if the instructor has custom privileges. 
- Also removed unneeded `waitForPageToLoad` to reduce timeout error probability.

Reply:
It seems I was wrong about why the test was failing. I tried to add a wait for the panel specifically but it did not work so well. The webdriver still throws missing element exception for the panel even though it is supposedly being waited for.

I am not too sure what is the real root cause but my hunch is that the part of the test with toggling repeats the edit and save action too quickly, causing the get and wait of the webdriver to interact with the old elements which results in errors. Adding a slight delay between the toggles greatly reduced the error rate.